server/src/drizzle/methods/User.ts

```diff
@@ -1,118 +1,76 @@
-import { User as TUser } from "../../db/schema";
 import * as jwt from 'jsonwebtoken'
-type ExtendedUser = {
-  save: (prisma: any) => Promise<ExtendedUser>;
-  toJSON: () => Partial<TUser>;
-  generateAuthToken: (prisma: any, jwtSecret: string) => Promise<string>;
-  generateResetToken: (
-    prisma: any,
-    jwtSecret: string,
-    clinetUrl: string,
-  ) => Promise<string>;
-};
+import { eq } from "drizzle-orm";
+import { createDb, } from "src/db/client";
+import { UserTable } from "src/db/schema";
 
-const User = <T>(prismaUser: T): T & ExtendedUser => {
-  if (!prismaUser) return;
-  return Object.assign(prismaUser, {
-    async save(prisma: any): Promise<ExtendedUser> {
-      const user = this;
+export class User {
```

Comment:
I like the idea of making this a class, I believe with the structure you set up here we can successfully handle all the computed values we had in mongoose.